packages/block-editor/src/components/block-tools/index.js

```diff
@@ -56,23 +61,25 @@ function selector( select ) {
  * insertion point and a slot for the inline rich text toolbar). Must be wrapped
  * around the block content and editor styles wrapper or iframe.
  *
- * @param {Object} $0                      Props.
- * @param {Object} $0.children             The block content and style container.
- * @param {Object} $0.__unstableContentRef Ref holding the content scroll container.
+ * @param {Object} $0                                   Props.
+ * @param {Object} $0.children                          The block content and style container.
+ * @param {Object} $0.__unstableContentRef              Ref holding the content scroll container.
+ * @param {string} $0.__experimentalBlockToolbarDisplay The display mode for the block toolbar.
  */
 export default function BlockTools( {
 	children,
 	__unstableContentRef,
+	__experimentalBlockToolbarDisplay = 'popover',
```

Comment:
Why do we need this, can we just rely on the existing setting within the block editor store instead?

Reply:
I had originally thought there was a reason we needed to pass this in to have full control over all the presentation states we need, but I think you're right that we can return to relying on the fixed toolbar settings.

Reply:
We do need a prop to at least "turn off" the toolbar entirely though, otherwise we'll have two toolbars in the DOM when we try to use our own top toolbar in the WordPress editor headers.

Reply:
I guess in my mind `hasFixedToolbar` means don't render the toolbar in block tools.

Reply:
So, for the storybook "box" example, would it look like this? 

```
<BlockEditorProvider
	value={ blocks }
	onInput={ updateBlocks }
	onChange={ updateBlocks }
	settings={ {
		hasFixedToolbar: true,
	} }
>
	<BlockToolbar />
	<BlockTools />
	<BlockCanvas height="100%" styles={ editorStyles } />
</BlockEditorProvider>
```

If that's the case, I think `hasBlockToolbar` is a more descriptive name. But it feels odd to have a setting in the wrapper called `hasBlockToolbar`, but you need to implement the toolbar within it anyways.

Reply:
> I guess in my mind hasFixedToolbar means don't render the toolbar in block tools.

https://github.com/WordPress/gutenberg/pull/56335/commits/ce1890d9d08454dc7574254c9865ecba282a1d79 implements this idea for the post editor. The main idea:
- Don't render `<BlockToolbar />` in `<BlockTools />` if hasFixedToolbar is on.
- On small screens at the top level `<Editor />` where `hasFixedToolbar` is set, force it to `true` if `! LargeViewport`.
- On small screens in `<Layout />`, render `<BlockTools />` if `! LargeViewport`.

This keeps the UI all the same for the post editor, but implements the concept.

Reply:
I think that make sense to me. 

Curious, what does `<BlockTools />` do in the example you shared above (the box example)?

Reply:
I agree the name of the settings is not great but I'm thinking that it's fine to avoid breaking changes.